src/main/java/org/hypertrace/gradle/code/style/CodeStylePlugin.java

```diff
@@ -36,26 +41,40 @@ private void configureFormatting(SpotlessExtension spotlessExtension) {
         });
 
     spotlessExtension.kotlinGradle(
-        format ->
-        {
-            try {
-                format
-                    .ktlint("0.50.0")
-                    .editorConfigOverride(
-                        new HashMap<String, Object>() {
-                          {
-                            put("indent_size", "2");
-                          }
-                        });
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
+        format -> {
+          try {
+            format.ktlint("0.50.0").editorConfigOverride(Map.of("indent_size", "2"));
+          } catch (IOException e) {
+            throw new RuntimeException(e);
+          }
         });
+
+    BufExtension bufExtension = project.getExtensions().getByType(BufExtension.class);
+    Provider<File> bufBinaryProvider =
+        project.provider(
+            () -> {
+              File binary =
+                  project
+                      .getConfigurations()
+                      .getByName(BufSupportKt.BUF_BINARY_CONFIGURATION_NAME)
+                      .getSingleFile();
+              if (!binary.canExecute()) {
+                binary.setExecutable(true);
+              }
+              return binary;
+            });
+    spotlessExtension.protobuf(
+        format ->
+            format
+                .buf(bufExtension.getToolVersion())
+                .pathToExe(bufBinaryProvider.get().getAbsolutePath()));
```

Comment:
so the provider part is right, but if pathToExe doesn't accept a provider we don't get the benefit we're looking for - we want gradle to be resolving the provider JIT so `.get()` is usually a red flag.

This actually is a closure already (the lambda from 67) it's just one that likely evaluates during config time rather than execution. So moving the resolution inside here appears to be the best we can do. Given that, we have the option of either removing the explicit provider and moving that code inside this lambda or leaving it as is. They should be equivalent, so up to you which you feel is more readable.

Reply:
Got it, I see what you mean now. I believe the spotless gradle plugin supports lazy configuration, so it shouldn't try to configure the protobuf plugin if the user is not formatting files. I'll revert my last commit that moved the resolution from this lambda to a provider.